### PR TITLE
Various docs improvements

### PR DESCRIPTION
**Pull request by miguno**

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -66,10 +66,10 @@ KSQL Server
     The KSQL server runs the engine that executes KSQL queries. This includes processing, reading, and writing data to and from
     the target Kafka cluster.
 
-    Servers can run in containers, virtual machines, and bare-metal machines. You can add or remove multiple servers in the
-    same resource pool to elastically scale query processing in or out. You can use different resource pools to achieve workload
-    isolation.
+    KSQL servers form KSQL clusters and can run in containers, virtual machines, and bare-metal machines. You can add or
+    remove multiple servers to/from the same KSQL cluster during live operations elastically scale query processing in
```
Comment (big-andy-coates) [inline docs/index.rst]: Is `multiple` superfluous, given it's already `servers` i.e. plural.

Comment (joel-hamill) [reply docs/index.rst]: +1

Comment (miguno) [reply docs/index.rst]: Updated

Comment (joel-hamill) [inline docs/index.rst]: you can replace `to/from` with `in`: `You can add or remove servers in the same KSQL cluster...`

File: docs/installation/index.rst [hunk 2]
```diff
@@ -3,17 +3,9 @@
 Installation and Configuration
 ------------------------------
 
-Interoperability
```
Comment (joel-hamill) [inline docs/installation/index.rst]: did the interoperability get moved somewhere else, or just deleted?

File: docs/installation/server-config/index.rst [hunk 3]
```diff
@@ -93,12 +95,9 @@ bootstrap.servers
...
-in the form ``host1:port1,host2:port2,...`` The default value in KSQL is ``localhost:9092``. For example, to change it to ``9095``
-by using the KSQL command line:
-
-.. code:: bash
+in the form ``host1:port1,host2:port2,...``. The default value in KSQL is ``localhost:9092``.
 
-    ksql> SET 'bootstrap.servers'='localhost:9095';
+.. important:: ``bootstrap.servers`` must not be set in the KSQL CLI via ``SET``. Define ``bootstrap.servers`` only in the KSQL server configuration file.
 
 For more information, see :ref:`Streams parameter reference <streams_developer-guide_required-configs>` and the :cp-javadoc:`Javadoc|clients/javadocs/org/apache/kafka/clients/consumer/ConsumerConfig.html#BOOTSTRAP_SERVERS_CONFIG`.
```
Comment (joel-hamill) [inline docs/installation/server-config/index.rst]: should this sentence be a part of `.. important`?

File: docs/installation/server-config/index.rst [hunk 4]
```diff
@@ -195,53 +200,50 @@ The default number of partitions for the topics created by KSQL. The default is
...
+    listeners=http://0.0.0.0:80
 
-----------------------------------
-Restricting Interactive KSQL Usage
-----------------------------------
+.. _restrict-ksql-interactive:
 
-KSQL supports locked-down deployment scenarios where you can restrict interactive use of the KSQL cluster.
+------------------------------------
+Headless, non-interactive KSQL Usage
```
Comment (joel-hamill) [inline docs/installation/server-config/index.rst]: shouldn't use commas in the title, and should be capitalized to match other titles (e.g. `Headless Non-interactive KSQL Usage`). on a side note - is the term `headless` known in this context?

File: docs/installation/server-config/index.rst [hunk 5]
```diff
@@ -195,53 +200,50 @@ The default number of partitions for the topics created by KSQL. The default is
...
+------------------------------------
 
-You can prevent interactive use of a KSQL cluster. For example, you want to allow a team of users to develop and verify
-their queries on a shared testing KSQL cluster. But when putting those queries to production you prefer to lock-down access
-to KSQL Servers, version-control the exact queries and storing them in a .sql file, and prevent users from interacting
-directly with the production KSQL cluster.
+KSQL supports locked-down, "headless" deployment scenarios where interactive use of the KSQL cluster is disabled.
+For example, you want to allow a team of users to develop and verify their queries interactively on a shared testing
```
Comment (joel-hamill) [inline docs/installation/server-config/index.rst]: `For example, to allow a team of users to develop and verify...`

File: docs/installation/server-config/index.rst [hunk 6]
```diff
@@ -195,53 +200,50 @@ The default number of partitions for the topics created by KSQL. The default is
...
+KSQL cluster. But when putting those queries to production you prefer to lock-down access to KSQL Servers,
```
Comment (joel-hamill) [inline docs/installation/server-config/index.rst]: `But when deploying those queries in your production environment, you want to lock-down access to KSQL Servers, version-control the exact queries, and store them in a .sql file. This will prevent users from interacting directly with the production KSQL cluster.`

**Comment by joel-hamill** [comment]
@miguno conflicts to resolve, then feel free to merge

File: docs/installation/index.rst (hunk 2 above)
Comment (miguno) [reply docs/installation/index.rst]: This got moved elsewhere IMHO.

File: docs/installation/server-config/index.rst (hunk 3 above)
Comment (miguno) [reply docs/installation/server-config/index.rst]: No

File: docs/installation/installing.rst [hunk 7]
```diff
@@ -111,15 +111,15 @@ Tip
...
+    and ``listeners``. You can also set any property the Kafka Streams API, the Kafka producer, or the Kafka consumer.
     For a description of common configurations and how to set them, see :ref:`configuring-ksql`.
 
     Here are the default settings:
 
     .. code:: bash
 
         bootstrap.servers=localhost:9092
-        listeners=http://localhost:8080
+        listeners=http://localhost:8088
```
Comment (miguno) [inline docs/installation/installing.rst]: Question (this is unrelated/not touched by the PR): I noticed that the new `docs/installation/server-config/config-reference.rst` file documents `listeners` as `ksql.listeners`.  However, it appears to me as if all other parts of the docs + the code still use the shorter `listeners`.  Is this intentional?

Comment (miguno) [reply docs/installation/installing.rst]: cc @joel-hamill

File: docs/installation/server-config/index.rst [hunk 8]
```diff
@@ -29,58 +31,58 @@ For example:
...
-For more information, see :ref:`ksql-parm-reference`.
+For more information, see :ref:`ksql-param-reference`.
 
 .. _configuring-ksql:
 
+-----------
 JMX Metrics
 -----------
 
 .. include:: ../../includes/ksql-includes.rst
```
Comment (miguno) [inline docs/installation/server-config/index.rst]: This is unrelated to this PR, but I noticed this while double-checking:  @joel-hamill it looks to me as if the latest `./docs/includes/ksql-includes.rst` file has JMX information at different start/end lines.  I think the correct numbers are now:

```
.. include:: ../../includes/ksql-includes.rst
:start-line: 325
:end-line: 332
```

Could you please confirm and follow-up with a separate PR if needed?

Comment (miguno) [reply docs/installation/server-config/index.rst]: And probably we need to double check the start/end lines for other includes, too.

**Comment by miguno** [comment]
@joel-hamill please review again to confirm the merge conflict resolutions I did were ok.

Also, I have run into two open questions for you (unrelated to this PR) while going through the PR again.  This might require follow-up PRs:

1. https://github.com/confluentinc/ksql/pull/1096#discussion_r178756167
2. https://github.com/confluentinc/ksql/pull/1096#discussion_r178757034

File: docs/installation/installing.rst (hunk 7 above)
Comment (joel-hamill) [reply docs/installation/installing.rst]: @rodesai ^ wydt

File: docs/installation/server-config/index.rst (hunk 8 above)
Comment (joel-hamill) [reply docs/installation/server-config/index.rst]: very good point. reviewing all now.

Comment (joel-hamill) [reply docs/installation/server-config/index.rst]: @miguno fixed https://github.com/confluentinc/ksql/pull/1112

File: docs/installation/server-config/index.rst [hunk 9]
```diff
@@ -29,70 +31,71 @@ For example:
...
     #. Start the KSQL with the predefined script specified via the ``--queries-file`` argument.
 
        .. code:: bash
 
             $ <path-to-confluent>/bin/ksql-start-server <path-to-confluent>/etc/ksql/ksql-server.properties \
-              --queries-file <path-to-queries-file>.sql
+              --queries-file /path/to/queries.sql
 
-Start the KSQL server in via the ``ksql-server.properties`` file
+To start the KSQL Server in headless, non-interactive configuration via the ``ksql.queries.file`` in the server
```
Comment (joel-hamill) [inline docs/installation/server-config/index.rst]: this line break after `server` is causing the definition list to lose its formatting: 
image

Should be a single non-broken line: `To start the KSQL Server in headless, non-interactive configuration via the ``ksql.queries.file`` in the server configuration file:`
